File: src/main/java/com/danyazero/utils/VoidType.java
package com.danyazero.utils;

import com.danyazero.model.Type;
import org.objectweb.asm.MethodVisitor;

public class VoidType implements Type<Void> {
    @Override
    public void postack(MethodVisitor mv, Void value) {
        throw new  UnsupportedOperationException();
    }

    @Override
    public void store(MethodVisitor mv, short index) {
        throw new  UnsupportedOperationException();
    }

    @Override
    public void load(MethodVisitor mv, short index) {
        throw new  UnsupportedOperationException();
    }

    @Override
    public void yield(MethodVisitor mv) {
        throw new  UnsupportedOperationException();
    }

    @Override
    public void aload(MethodVisitor mv) {
        throw new  UnsupportedOperationException();
    }

    @Override
    public void astore(MethodVisitor mv) {
        throw new  UnsupportedOperationException();
    }

    @Override
    public short getSize() {
        return 0;
    }
}

File: src/main/kotlin/type/VoidType.kt
package com.danyazero.type

import com.danyazero.model.Type
import org.objectweb.asm.Label
import org.objectweb.asm.MethodVisitor

class VoidType : Type<Void> {

    override fun postack(mv: MethodVisitor, value: Void) {
        throw UnsupportedOperationException()
    }

    override fun store(mv: MethodVisitor, index: Short) {
        throw UnsupportedOperationException()
    }

    override fun load(mv: MethodVisitor, index: Short) {
        throw UnsupportedOperationException()
    }

    override fun yield(mv: MethodVisitor) {
        throw UnsupportedOperationException()
    }

    override fun aload(mv: MethodVisitor) {
        throw UnsupportedOperationException()
    }

    override fun astore(mv: MethodVisitor) {
        throw UnsupportedOperationException()
    }

    override fun equal(mv: MethodVisitor, jumpTarget: Label) {
        //TODO VoidType equals operation
    }

    override fun getSize(): Short {
        return 1
    }
}